wagtail/admin/widgets.py

```diff
@@ -379,6 +382,18 @@ def show(self):
     @cached_property
     def dropdown_buttons(self):
         button_hooks = hooks.get_hooks(self.hook_name)
-        return sorted(itertools.chain.from_iterable(
-            hook(self.page, self.page_perms, self.is_parent, self.next_url)
-            for hook in button_hooks))
+
+        buttons = []
+        for hook in button_hooks:
+            if accepts_kwarg(hook, 'next_url'):
+                button = hook(self.page, self.page_perms, self.is_parent, self.next_url)
```

Comment:
This is a bit deceptive... the hook actually returns an iterable of buttons (one for every `yield` in the hook function - there may be multiple) and the `itertools.chain.from_iterable` takes care of turning this iterable of iterables into a final sequence that we can sort. As a result, for someone casually reading this code, this change is inadvertently making the `itertools` line even more voodoo magic than it was before :-)

This could be rectified by renaming `button` to `hook_result` or similar - however, since the 'fancy' code is really just going out of its way to avoid building an intermediate list - which we are now doing - I think it'd be better to turn it fully into non-fancy code...

* this line (and the corresponding one in the 'else' branch) becomes `buttons.extend(hook(self.page, self.page_perms, self.is_parent, self.next_url))` - i.e. append everything returned by the hook to the end of `buttons`
* instead of the `itertools` incantation, the ending just becomes
  ```
  buttons.sort()
  return buttons
  ```